shared/settings/account/index.tsx

```diff
@@ -33,14 +33,24 @@ export const SettingsSection = ({children}: {children: React.ReactNode}) => (
     {children}
   </Kb.Box2>
 )
-const AddButton = (props: {disabled: boolean; kind: 'phone number' | 'email'; onClick: () => void}) => {
+
+type AddButtonProps = {
+  disabled: boolean
+  kind: 'phone number' | 'email'
+  onClick: () => void
+  waitingKey?: string
+}
+const AddButton = (props: AddButtonProps) => {
+  const waiting = props.waitingKey ? Container.useAnyWaiting(props.waitingKey) : undefined
```

Comment:
You could use `WaitingButton` (with the waiting key) here to get this behavior for free. Not a big deal though